lib/Segments/WP.php

```diff
@@ -71,15 +71,13 @@ private function createOrUpdateSubscriber($currentFilter, $wpUser, $subscriber =
     if (empty($wpUser->first_name) && empty($wpUser->last_name)) { // phpcs:ignore Squiz.NamingConventions.ValidVariableName.NotCamelCaps
       $firstName = $wpUser->display_name; // phpcs:ignore Squiz.NamingConventions.ValidVariableName.NotCamelCaps
     }
-    $signupConfirmationEnabled = SettingsController::getInstance()->get('signup_confirmation.enabled');
-    $status = $signupConfirmationEnabled ? Subscriber::STATUS_UNCONFIRMED : Subscriber::STATUS_SUBSCRIBED;
     // subscriber data
     $data = [
       'wp_user_id' => $wpUser->ID,
       'email' => $wpUser->user_email, // phpcs:ignore Squiz.NamingConventions.ValidVariableName.NotCamelCaps
       'first_name' => $firstName,
       'last_name' => $lastName,
-      'status' => $status,
+      'status' => SubscriberEntity::STATUS_UNSUBSCRIBED,
```

Comment:
@lysyjan it looks like the WC user signup is handled.
But because of this change, if a WP user signs up via the WordPress user registration form, and they check the "I want to subscribe" checkbox - do they get handled as if they want to subscribe? (added as unconfirmed, sent a confirmation email, scheduled welcome emails)

Based on the code below it doesn't appear to be the case, but maybe the consent check is handled elsewhere?

Reply:
@flakas Everything works as we expect. I have tested it again. I have tried the welcome email, the confirmation email, and I have checked the subscriber status.

The checkbox from the registration form is managed in the [Registration.php](https://github.com/mailpoet/mailpoet/blob/master/lib/Subscription/Registration.php#L58-L74).